src/celeritas/track/detail/TrackSortUtils.hh

```diff
@@ -60,6 +60,17 @@ template<>
 void partition_tracks_by_status(
     CoreStateData<Ownership::reference, MemSpace::device> const& states);
 
+//---------------------------------------------------------------------------//
+template<MemSpace M>
+void sort_tracks_by_action_id(
+    CoreStateData<Ownership::reference, M> const& states);
+template<>
+void sort_tracks_by_action_id(
+    CoreStateData<Ownership::reference, MemSpace::host> const& states);
+template<>
+void sort_tracks_by_action_id(
+    CoreStateData<Ownership::reference, MemSpace::device> const& states);
```

Comment:
Since we specialize both of the `MemSpace` values, I don't think there's any advantage using templates versus overloading...